rfc/monorepo.md

```diff
@@ -0,0 +1,557 @@
+
+The initial version of a numbered release is considered to have patch version zero, even if not explicitly stated for brevity.  Unqualified numbered releases refer to the latest patch of that numbered release.
+
+Significant new feature work should result in a new release number.  Bugfixes, security fixes, or other important non-breaking, non-feature changes may be introduced in patch releases.  Small updates to existing functionality may be considered for patch releases if the changes are additive and non-breaking - this is left to the judgement of the reviewer and approver, along with the release manager.
+
+All patch releases (`<year>.<number>.<patch>`) should be fully compatible with that version's initial release, and all previous patches to that release number.  Patch releases should be fully non-breaking, from external API all the way down to public function signatures in libraries.  Exceptions can be made to mitigate critical bugs or security issues.
+
+Features and API annotated with `@Alpha` or `@Beta` annotations may change at any time.  However, it is desirable to minimize breakage in patch releases when possible, particularly if annotated as `@Beta`.  This is also left to the judgement of the reviewer, approver, and release manager.
+
+No contract is given between numbered releases (`<year>.<number>`) - these may break, add, remove, or otherwise change any functionality deemed necessary by the project.
+
+#### Release Timing and Cadence
+
+Release timing and cadence is not addressed by this RFC.  When a new release is made during a calendar year, the release number increases by one.  When a patch release is made, the patch number increases by one.
+
+The release manager retains the authority to release as they see fit.  Backports to older versions will be judgements made based on the criticality of the issue, risk to a release's stability, and the difficulty of backporting the change itself.
+
+#### Release Branch Names
+
+Release branches will have a naming scheme similar to today, but date-oriented.  A release branch is prefixed by `release-`, appends the release version, and is suffixed with `.x`, e.g. `release-2020.3.x`.
+
+This branch will produce containers, libraries, and other artifacts for that numbered release.
+
+#### Artifact Versioning Scheme
+
+Spinnaker library and container versions from release branches will mirror the release version (e.g. `2023.3.4`).  This should be defined in the root `gradle.properties` file, and used throughout included builds and subprojects to publish their respective artifacts.
+
+From `main`, library and container versions will be published as `main-<gha_run_id>`, e.g. `main-4041166540`.  Github Actions run IDs always increase, so this functions identically to timestamp-based snapshot versions with two key benefits:
```

Comment:
I could see doing commit hash to allow tracking jars to a specific commit.  GHA Run id isn't bad and does allow easy view of order.

Reply:
GHA run ID does directly map to a commit, and we can include that in jar metadata if we want the sha directly.  I don't usually prefer having a sha in the jar name, since it makes the library more difficult to consume.  Gradle ordering with a sha can be very non-obvious, and if we happen to get an all-numeric short sha, it gets weirder still.